### DISCUSSION TITLE
⭐ New Release: 3.3.0 RC 1

### DISCUSSION THREAD

**Discussion by Xanashi** [Xanashi/Icaros#30] [Releases]
Good Afternoon Everyone,

I had a couple showstoppers, that I really wanted to get fixed before releasing the final version of 3.3.0,
but they ended up taking a lot longer to fix and test, than I had originally anticipated, so that's basically why 3.3.0
hasn't been released yet. This is also why this build is still only a release candidate, as I want to make sure
I haven't introduced any new bugs with these fixes before making this build final.

And if everything (hopefully) works the same or better than Beta 3, this build will be released as the final build of 3.3.0.

I have a few notes on some of the below fixes, but to avoid making this post too long, I'll post them below in a separate post.

Also, a while ago someone asked me if I could move the hosting of the installer and zip to Github, as it may help to get Icaros
added to package managers such as winget. So from now on, you can find the latest Icaros releases on the [official Icaros github page](https://github.com/Xanashi/Icaros/releases/).

I hope you all enjoy the new releases!

_______________________________________________________________________________________________________

**[Icaros v.3.3.0 RC 1]**

- Implemented a way for Icaros to recover slightly corrupted cache files
- Improved how IcarosConfig handles permission issues for the Icaros Cache
- Added support for thumbnailing AVIF with alpha channels
- Fixed a lot of small Cache issues
- Fixed Cache Indexer returning an 0x14 error when parsing long filenames
- Fixed a couple of rare issues that affected thumbnailing of some files
- Added a workaround for an issue that prevented the Icaros Thumbnail Provider from working on Vista
- Installer is now localized (Thank you bovirus for the implementation!)
- Installer has been updated with a slightly more modern look
- Added Romanian localization (Thank you Cătălin Popescu! ^__^)
- Updated Italian localization (Thank you bovirus! ^__^)
- IcarosConfig now uses %SystemRoot% instead of the hardcoded C:\Windows path when referencing the WOW64 regsvr32.exe
- Updated LibDav1d to 1.0.0
- Updated GCC
- Updated FFmpeg
<br>

Icaros 3.3.0 RC 1 Installer: [Download](https://github.com/Xanashi/Icaros/releases/download/v3.3.0rc1/Icaros_v3.3.0_RC1.exe)
Icaros 3.3.0 RC 1 Portable: [Download](https://github.com/Xanashi/Icaros/releases/download/v3.3.0rc1/Icaros_v3.3.0_RC1.zip)

**Comment by Xanashi** [comment]
## Cache Recovery

During the testing of the 3.3.0 Beta builds, a couple of users reported a pretty obscure bug that
caused their Icaros Cache to no longer work. I couldn't quite figure out what exactly had happened,
until a generous tester send me their cache files.
Apparently their cache had gotten slightly corruption, because Icaros had been forcefully terminated
while trying to update their cache files.

Instead of asking users to recreate their caches from scratch in these cases, I decided to implement
a recovery system into the Icaros Cache, allowing it to fix these (hopefully rare) cases where the
cache somehow ends up in an invalid state.

That being said, adding a recovery system to the Icaros Cache was a bit of an intimidating task,
and of course ended taking longer than I wanted it to, because I wanted to cover as many potential
scenarios as possible. But now it's implemented and should solve issues like the one described above 
automatically without the user ever knowing. 
<br/>

## Cache Permission Issues

I've tried to improve the way IcarosConfig accesses the Icaros Cache directory in this release.
The thing is, the Icaros Cache only needs user access to the Icaros Cache directory, but IcarosConfig
needs admin access. This is not a problem on most users systems, but those few that have a more limited
system permission setup, often run into an error message when dealing with the Cache.

Hopefully the latest changes will fix some of the bugs these users has run into, however it is just not
easy to replicate these various permission setups, without having the complete knowledge of what is going on
other peoples systems. So let me know, if you still experience cache permission issues with this new release.
<br/>

## Vista

I recently had a report, that the latest beta version of Icaros no longer worked on Windows Vista.
I looked into the issue, and it turned out that libdav1d had added some Windows API calls that wasn't
compatible with Vista. Like many modern libraries, libdav1d officially only supports Windows 7 and above.
I was able add a workaround this time around, but to ensure Icaros can make use of quality libraries
like libdav1d and FFmpeg going forward, this will probably be the last build, where Icaros officially supports Windows Vista.

This doesn't mean future builds of Icaros won't work on Vista, but it does mean that if libraries like FFmpeg or libdav1d
implements new features in the future, that are completely incompatible with Vista, I will not be trying to work around them
to make Icaros work on Vista again.

I hope that makes sense. Otherwise let me know below.
<br/>

**Comment by Witchilich** [comment]
https://github.com/microsoft/winget-pkgs/pull/92161
winget package should be approved soon. 

Update: Its approved. It can be installed from the command line with `winget install Xanashi.Icaros.Beta`.

**Reply by Xanashi** [reply]
Thank you so much Witchilich! ❤️ 
That's awesome.
